### PR TITLE
Add r and rstudio images

### PR DESCRIPTION
**Pull request by riazarbi**
This PR does the following:

1. Remove rsession image (broken anyway)
2. Add an r and rstudio image.
3. Update the README.md

The r image has a working R installation. It also checks what the Ubuntu system requirements for the top 100 downloaded packages on CRAN over the last month are, and installs those. It does not install the actual R packages. This keeps the image slim, but makes per-project package installs likely to succeed.

The Rstudio image has Jupyter, Rstudio and VScode installed, and is configured to launch Jupyter in the same fashion as jupyter-ide. 

The python_minimal image is 2.92GB. The r image is 4.44GB, and the Rstudio image is 6.82GB. This compares favourably with the Jupyter stack - jupyter-ide is 8.22GB.

This PR has not modified any jenkins configs, so these images won't build automatically, you need to build them manually until you incorporate them into CI.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,13 +8,13 @@ The current structure of these image tags (and the repo):
...
     └── drivers
-        ├── python_minimal
-        ├── jupyter-ide
-        ├── python
-        │   └── jupyter-k8s
-        └── r_minimal
+        └── python_minimal
+            ├── jupyter-ide
+            |   └── python
+            |       └── jupyter-k8s
```
Comment (Gordonei) [inline README.md]: `jupyter-ide` is at the same level as `python`

```suggestion
            |   ├── jupyter-ide
            |   └── python
            |       └── jupyter-k8s
```

File: README.md [hunk 2]
```diff
@@ -24,18 +24,23 @@ Overview of the tags:
...
 docker exec -it -rm cityofcapetown/datascience:python bash
-```
\ No newline at end of file
+```
+
+To launch a jupyter-based web server from `jupyter-ide`, `jupyter-k8s` or `rstudio` enter something like the following:
+
+```bash
+docker run -it --rm -p 8888:8888 cityofcapetown/datascience:{jupyter-ide,jupyter-k8s,rstudio}
+```
```
Comment (Gordonei) [inline README.md:46]: thanks, a useful addition to the docs

File: base/drivers/python_minimal/r/rstudio/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+ # Install Rstudio
+ && wget --quiet https://download2.rstudio.org/server/jammy/amd64/rstudio-server-2024.04.2-764-amd64.deb  \
+ && gdebi -n rstudio-server-2024.04.2-764-amd64.deb \
+# Clean up
+ && rm -rf /var/lib/apt/lists/* \
+ && rm -rf /tmp/downloaded_packages \
+ && rm -rf rstudio-server-2024.04.2-764-amd64.deb \
+# Install VSCode
+ && curl -fsSL https://code-server.dev/install.sh | sh \
+ && pip3 install jupyter-vscode-proxy
```
Comment (Gordonei) [inline base/drivers/python_minimal/r/rstudio/Dockerfile]: let me check with the team, but don't know if anyone is actively doing R work outside of VS Code

How much space it adds to the image?

**Review by Gordonei — CHANGES_REQUESTED** [review]
Your commit(ment) drives the City forward!

File: base/drivers/python_minimal/r/rstudio/Dockerfile (hunk 3 above)
Comment (riazarbi) [reply base/drivers/python_minimal/r/rstudio/Dockerfile]: With VSCode - 6.82Gb
Without VSCode - 5.82Gb

**Comment by riazarbi** [comment]
There will be some setup required for the R users when they spawn this image for the first time. In particular - 

1. Set up proxy so that the IDE can hit CRAN and optionally apt to install packages
2. `quarto install tinytex` will install tinytex to $HOME. This hasn't been done because $HOME is mounted to it would get overwritten is installed prior.

There is pretty good integration for `renv` in Rstudio.

File: base/drivers/python_minimal/r/rstudio/Dockerfile (hunk 3 above)
Comment (Gordonei) [reply base/drivers/python_minimal/r/rstudio/Dockerfile]: Thinking about it more, what I'll rather do is move `jupyter-ide` below `r_minimal`, and add the R VS Code extension there, so remove it from here for now, please.

The idea was always that `jupyter-ide` would be for preparing stuff for production, hence is usually the time when people want to use IDEs

Comment (riazarbi) [reply base/drivers/python_minimal/r/rstudio/Dockerfile]: Done

**Comment by riazarbi** [comment]
Ok, I think you are good to go.

**Comment by Gordonei** [comment]
LGTM

**Pull request merged**